arelle/plugin/validate/NL/rules/fr_nl.py

```diff
@@ -227,7 +227,7 @@ def rule_fr_nl_1_05(
     """
     for doc in val.modelXbrl.urlDocs.values():
         if doc.type == ModelDocument.Type.INSTANCE:
-            if 'UTF-8' != doc.xmlDocument.docinfo.encoding:
+            if 'UTF-8' != doc.documentEncoding.upper():
```

Comment:
Out of curiosity, why doesn't `doc.xmlDocument.docinfo.encoding` work, and why is `doc.documentEncoding` the correct thing to use?

(As an aside, I don't know understand what [this logic](https://github.com/Arelle/Arelle/blob/0975615971ef93de11d19b59bc5cf8750bb1b9d1/arelle/XmlUtil.py#L160-L161) is doing.)

Reply:
> Out of curiosity, why doesn't `doc.xmlDocument.docinfo.encoding` work, and why is `doc.documentEncoding` the correct thing to use?

Depending on the version of lxml it will detect the encoding as ascii if the file content is limited to ascii characters even if the XML declaration is utf-8.
 
> (As an aside, I don't know understand what [this logic](https://github.com/Arelle/Arelle/blob/0975615971ef93de11d19b59bc5cf8750bb1b9d1/arelle/XmlUtil.py#L160-L161) is doing.)

I believe that's a legacy pattern for declaring encoding in python scripts (not sure why Arelle needs that for loading files).

Reply:
> Depending on the version of lxml it will detect the encoding as ascii if the file content is limited to ascii characters even if the XML declaration is utf-8.

I see.  Is there some lxml bug/release note, or how did we discover that?  A comment.. somewhere.. (maybe in `def encoding`? or on `documentEncoding`?) that explains why we're this should be using rather than`xmlDocument.docinfo.encoding` might be useful.

Reply:
I added a comment. Arelle [strips the XML declaration](https://github.com/Arelle/Arelle/blob/0975615971ef93de11d19b59bc5cf8750bb1b9d1/arelle/ModelDocument.py#L157) before parsing the file contents with lxml, so lxml is detecting the encoding without the declaration. I tried changing that, but including the declaration (which lxml then requires the content to be bytes) breaks many many tests and appears to require a significant rewrite.

Reply:
Thanks.  That `stripDeclaration` seems to wrongly allow `<?xml encoding='utf-8'?>` (missing version) or even `<?xml invalid='invalid'?>`.  It seems like the code should be updated to always pass bytes to the XML parser.  I agree that looks non-trivial.